CuBeanzQueryable.sol

```diff
@@ -0,0 +1,251 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+error NotOwner();
+error MaxTeamMints();
+error RoundSoldOut();
+error SaleNotStarted();
+error InvalidValue();
+error MaxMints();
+error ContractMint();
+/*
+
+░█████╗░██╗░░░██╗██████╗░███████╗░█████╗░███╗░░██╗███████╗
+██╔══██╗██║░░░██║██╔══██╗██╔════╝██╔══██╗████╗░██║╚════██║
+██║░░╚═╝██║░░░██║██████╦╝█████╗░░███████║██╔██╗██║░░███╔═╝
+██║░░██╗██║░░░██║██╔══██╗██╔══╝░░██╔══██║██║╚████║██╔══╝░░
+╚█████╔╝╚██████╔╝██████╦╝███████╗██║░░██║██║░╚███║███████╗
+░╚════╝░░╚═════╝░╚═════╝░╚══════╝╚═╝░░╚═╝╚═╝░░╚══╝╚══════╝
+
+Dev: Shimazu Bohoro
+*/
+
+
+import "erc721a/contracts/extensions/ERC721AQueryable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+
+
+contract CuBeanzQueryable is  ERC721AQueryable , Ownable, ReentrancyGuard {
+    using Strings for uint256;
+    using ECDSA for bytes32;
+
+    //@dev mint constraints
+    uint16 public teamMints;
+    uint16 maxTeamMints = 444;
+    uint16  roundOneSupply = 2222;
+    uint16  roundTwoSupply = 1778;
+    uint16 public roundOneMints;
+    uint16 public roundTwoMints;
+
+    uint8 maxRoundOneMints = 2;
+    uint8 maxRoundTwoMints = 1;
+    uint8 public maxAttacks  = 1;
+    
+
+    uint80 public burnRate = 10000 ether;
+    uint80 public cubePrice = 22000 ether; 
```

Comment:
Since there is no onlyOwner setter for:
- `burnRate`
- `cubeRate`

these variables can be marked as constant to save gas.